code/game/objects/items/weapons/grenades/grenade.dm

```diff
@@ -61,23 +71,26 @@
 		T.hotspot_expose(700,125)
 
 /obj/item/grenade/attackby(obj/item/W as obj, mob/user as mob)
-	if(isScrewdriver(W))
-		switch(det_time)
-			if (1)
-				det_time = 10
-				to_chat(user, "<span class='notice'>You set the [name] for 1 second detonation time.</span>")
-			if (10)
-				det_time = 30
-				to_chat(user, "<span class='notice'>You set the [name] for 3 second detonation time.</span>")
-			if (30)
-				det_time = 50
-				to_chat(user, "<span class='notice'>You set the [name] for 5 second detonation time.</span>")
-			if (50)
-				det_time = 1
-				to_chat(user, "<span class='notice'>You set the [name] for instant detonation.</span>")
-		add_fingerprint(user)
+	if(!emagged)
+		if(isScrewdriver(W))
+			switch(det_time)
+				if (1)
+					det_time = 10
+					to_chat(user, "<span class='notice'>You set the [name] for 1 second detonation time.</span>")
+				if (10)
+					det_time = 30
+					to_chat(user, "<span class='notice'>You set the [name] for 3 second detonation time.</span>")
+				if (30)
+					det_time = 50
+					to_chat(user, "<span class='notice'>You set the [name] for 5 second detonation time.</span>")
+				if (50)
+					det_time = 1
+					to_chat(user, "<span class='notice'>You set the [name] for instant detonation.</span>")
+	else
+		to_chat(user, SPAN_WARNING("\The [src] refuses the change!"))
+	add_fingerprint(user)
```

Comment:
Switch the order. Right now if you use ANY item on it it will do the emag warning.
I am also not so sure on the whole idea of emagging a grenade.

Reply:
done